### PR TITLE
WIP: feat: uggla feat/factorize exporter (fixing conflicts)

### PR DESCRIPTION
**Pull request by bpetit**

### CODE REVIEW

**Review by uggla — COMMENTED** [review]
Quick review. Sounds good to me.

File: docker-compose/prom-dev/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,5 @@
+FROM prom/prometheus
```
[the remark this answers is not in the document]
Comment (bpetit) [reply docker-compose/prom-dev/Dockerfile:1]: you mean for riemann ?

File: src/exporters/stdout.rs [hunk 2]
```diff
@@ -11,6 +11,9 @@ pub struct StdoutExporter {
 }
 
 impl Exporter for StdoutExporter {
+    // fn manage_metric<T>(&self, client: T, data: &Vec<Metric>) {
```
[the remark this answers is not in the document]
Comment (bpetit) [reply src/exporters/stdout.rs:14]: you're right

File: src/exporters/riemann.rs [hunk 3]
```diff
@@ -1,495 +1,175 @@
...
                 .as_secs() as i64,
         );
-        event.set_ttl(ttl);
-        event.set_host(hostname.to_string());
-        event.set_service(service.to_string());
-        event.set_state(state.to_string());
-        event.set_tags(protobuf::RepeatedField::from_vec(tags));
+        event.set_ttl(metric.ttl);
+        event.set_host(metric.hostname.to_string());
+        event.set_service(metric.name.to_string());
```
[the remark this answers is not in the document]
Comment (bpetit) [reply src/exporters/riemann.rs:60]: I think I did: [https://github.com/hubblo-org/scaphandre/pull/96/files/3307fc688a4d7ed5c22bb66240da3af992fe330f#diff-a4d0fc55e2bba58ed7c21132fbd2ba58552a5035b7244c057c16440a3f8e211cR177](https://github.com/hubblo-org/scaphandre/pull/96/files/3307fc688a4d7ed5c22bb66240da3af992fe330f#diff-a4d0fc55e2bba58ed7c21132fbd2ba58552a5035b7244c057c16440a3f8e211cR177)

Did I miss something ?

File: docker-compose/prom-dev/Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (uggla) [reply docker-compose/prom-dev/Dockerfile:1]: yep sorry riemann... 😝

File: src/exporters/riemann.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (uggla) [reply src/exporters/riemann.rs:60]: I think yes.
If I'm not wrong you remove:
```rust
            let metric_name = format!(
                "{}_{}_{}",
                "",
                pid.to_string(),
                exe
            );
```
from the mod.rs file. But I can't find something similar in riemann.rs.
I think something like:
```rust
if metric.name == "scaph_process_power_consumption_microwatts"{
  let metric_name = format!(
                "{}_{}_{}",
                "scaph_process_power_consumption_microwatts",
                pid.to_string(),  --> this is the idea code is incorrect here (need to be extracted from the attributes above
                exe --> this is the idea code is incorrect here (need to be extracted from the attributes above
            );
}
```

[the remark this answers is not in the document]
Comment (bpetit) [reply src/exporters/riemann.rs:60]: It's ont highlighted as it was like that in riemann.rs before your PR, but I did restore those few lines, line 177 in the current branch. Are we talking about the same thing ?

File: docker-compose/prom-dev/Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (bpetit) [reply docker-compose/prom-dev/Dockerfile:1]: do you have in mind the kind of query we could run on riemann from the terminal to test the metrics are coming in ?

**Comment by bpetit** [comment]
for the notice, the extra tests : [https://github.com/hubblo-org/ansible-scaphandre/blob/master/validate_gh.yml#L49](https://github.com/hubblo-org/ansible-scaphandre/blob/master/validate_gh.yml#L49)

File: src/exporters/riemann.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (uggla) [reply src/exporters/riemann.rs:60]: hum I guess we don't understand each other. Will you be available for a quick chat ? Jitsi or gitter ?

[the remark this answers is not in the document]
Comment (bpetit) [reply src/exporters/riemann.rs:60]: Let's discuss that on gitter when you have some time :)

**Pull request merged**